grammar/haxe.bnf

```diff
@@ -527,20 +537,18 @@ left shiftExpression ::= shiftOperator additiveExpressionWrapper
 private additiveExpressionWrapper ::= multiplicativeExpressionWrapper additiveExpression*
 left additiveExpression ::= ('+' | '-') multiplicativeExpressionWrapper
 
-private multiplicativeExpressionWrapper ::= assignExpressionWrapper multiplicativeExpression*
-left multiplicativeExpression ::= ('*' | '/' | '%') assignExpressionWrapper
+private multiplicativeExpressionWrapper ::= prefixExpression multiplicativeExpression*
+left multiplicativeExpression ::= ('*' | '/' | '%') (prefixExpression | suffixExpressionWrapper)
 
 /* Technically, assignExpressions don't have higher priority than everything else.
```

Comment:
Not at all, this rule indeed gives us right order an parsing without errors constructions with nested assign expressions, look at ```testData/parsing/expressions/ComplexExpression``` ```.hx``` and ```.txt```